### ISSUE TITLE
The registration process fails when the device's time changes, until the device is rebooted.

### ISSUE THREAD

**Issue by alexiagonzalez** [BelledonneCommunications/linphone-android#414]
Good morning, 

First of all, thanks for your great job, Linphone is a very useful tool.

We've observed that when device's hour changes, the previously registered user fails, and the app changes to fail on register status. Normal app behavior is not recovered until the device is rebooted.

This causes that, if a user/device changes the time (manually, or maybe automatically on travels to other countries), the calls don't be received anymore until the user reboots its mobile.

Have you noticed that? Have you any idea to solve this problem, or any future intention to consider revise that issue?

We hope so. Thanks in advance, and best regards from Spain.

**Comment by Viish** [comment]
Thanks, we'll take a look at this.

**Comment by alexiagonzalez** [comment]
Hi again, 

Any news about this problem? We're working in an app and this would be the last pending issue to finish it.

We know you are probably busy but any help will be very appreciated.

Thanks a lot.

**Comment by Viish** [comment]
On the development branch of the release to come I don't have this issue.
Which version are you using ?

**Comment by alexiagonzalez** [comment]
Thanks a lot for your answer Viish.

We have observed this problem also in last Linphone version (at Google Play) 4.0.1, but we're using an older Linphone version as base for our project (in our own repository, not in Linphone repository), so we think this could be not a recent Linphone problem.

The issue is related to SIP registering. We were wondering if its origin could be in the library that handles SIP processes.

I don't know if this would be the right place to report a case as ours, however the issue is happening in published Linphone app too, so we try to report it here first. Any help would be appreciated if you know where the issue origin could be found, or if you think this could be fixed soon.

Thanks again Viish, and best regards from Spain.

**Comment by alexiagonzalez** [comment]
Hi again Viish, 

We have implemented a possible solution for this problem, it seems to fix the issue at least in our app. We comment it below.

We observed that the problem only appeared when the time change was performed with the app in background. If the app was out of memory, the problem didn't happen.

As the problem disappears when the device is rebooted, we supposed the issue could affect the service, that is also restarted in that case, so we're now restarting the service every time the app detects a time change. To detect time changes at the app, we have used a simple class that is informed by Android of that event.

`import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class TimeChangedReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        // Actions to do on a device time change.

        // On device time changes, the service could be corrupted and the registration fails.
        // To avoid that issue restart the service if possible.
        System.out.println("WARNING: DEVICE TIME CHAGE DETECTED. RESTARTING SERVICE.");
        restartService();
    }
}`

restartService() simply stops service if it's active and restarts it again.

It's also necessary to add something like this to the AndroidManifest file:

`        <receiver android:name=".TimeChangedReceiver">
            <intent-filter>
                <action android:name="android.intent.action.TIME_SET" />
            </intent-filter>
        </receiver>`

We don't know if that is the better solution, or if it could be problematic for some reason, but it's our workaround for the moment. Maybe it can be useful for someone.

Thanks again. Best regards.
